src/openrct2/interface/window.c

```diff
@@ -2582,3 +2598,86 @@ bool window_is_visible(rct_window* w)
     w->viewport->visibility = VC_VISIBLE;
     return true;
 }
+
+/**
+ *
+ *  rct2: 0x006E7499
+ * left (ax)
+ * top (bx)
+ * right (dx)
+ * bottom (bp)
+ */
+void window_draw_all(rct_drawpixelinfo *dpi, sint16 left, sint16 top, sint16 right, sint16 bottom)
+{
+    rct_drawpixelinfo windowDPI;
+    windowDPI.bits = dpi->bits + left + ((dpi->width + dpi->pitch) * top);
+    windowDPI.x = left;
+    windowDPI.y = top;
+    windowDPI.width = right - left;
+    windowDPI.height = bottom - top;
+    windowDPI.pitch = dpi->width + dpi->pitch + left - right;
+    windowDPI.zoom_level = 0;
+
+    for (rct_window *w = g_window_list; w < gWindowNextSlot; w++) {
+        if (w->flags & WF_TRANSPARENT) continue;
+        if (right <= w->x || bottom <= w->y) continue;
+        if (left >= w->x + w->width || top >= w->y + w->height) continue;
+
+        window_draw(&windowDPI, w, left, top, right, bottom);
+    }
+}
+
+rct_viewport * window_get_previous_viewport(rct_viewport * current)
+{
+    bool foundPrevious = (current == NULL);
+    rct_window * window = gWindowNextSlot;
+
+    while (true)
+    {
+        window--;
+        if (window < g_window_list)
```

Comment:
In other parts of this change you check for `gWindowNextSlot == NULL`, and indeed `gWindowNextSlot` is initialised to `NULL`. If you happen to initialise `window` with `gWindowNextSlot == NULL`, then `window--` in line above will make it point at some high, but invalid offset in memory.

You should add a check for `window != NULL` as a precondition to entering this loop.